lib/instance-detail-helper.js

```diff
@@ -264,9 +264,33 @@ function listDatabases(results, done) {
   adminDb.command(spec, {}, function(err, res) {
     if (err) {
       if (isNotAuthorized(err)) {
-        // we caught this further up already and really should never get here!
-        debug('listDatabases failed. returning empty list []');
-        done(null, []);
+        // eslint-disable-next-line no-shadow
+        adminDb.command({connectionStatus: 1, showPrivileges: 1}, {}, function(err, res) {
+          if (err) {
+            done(err);
+            return;
+          }
+          const privileges = (res.authInfo || {}).authenticatedUserPrivileges;
+          if (privileges === undefined) {
+            done(null, []);
```

Comment:
Just noticed, add return here to make sure that the callback after if statement won't be called.

Reply:
Fixed